src/lib/index.js

```diff
@@ -437,6 +437,27 @@ lib.castOption = function(trace, ptNumber, astr, fn) {
     }
 };
 
+/** Extract option from calcdata item, correctly falling back to
+ *  trace value if not found.
+ *
+ *  @param {object} calcPt : calcdata[i][j] item
+ *  @param {object} trace : (full) trace object
+ *  @param {string} calcKey : calcdata key
+ *  @param {string} traceKey : aka trace attribute string
+ *  @return {any}
+ */
+lib.extractOption = function(calcPt, trace, calcKey, traceKey) {
+    var calcVal = calcPt[calcKey];
+    if(calcVal || calcVal === 0) return calcVal;
+
+    // fallback to trace value,
+    //   must check if value isn't itself an array
+    //   which means the trace attribute has a corresponding
+    //   calcdata key, but its value is falsy
+    var traceVal = lib.nestedProperty(trace, traceKey).get();
+    if(!Array.isArray(traceVal)) return traceVal;
```

Comment:
Nice, I like it. Thanks for accepting `0`.

> not super happy with the name

meh, it's a very specific concept that we probably don't want to go into any more detail with in the name. Seems fine to me.

> not super happy about it being on Lib

It definitely belongs next to `castOption` but yeah, if we make a `common/` that might be more appropriate for these.

Also a style question: I just noticed all over this file we start the docs immediately after `/**`, I thought the convention was to leave that first line blank and start on the next line?

Reply:
![image](https://user-images.githubusercontent.com/6675409/31197857-74841f80-a920-11e7-9a27-d63742b340f2.png)

from http://usejsdoc.org/howto-commonjs-modules.html

So, I think both are acceptable.

Reply:
perhaps overly pedantic, but I read that as `/** Single-line description */` or
```
/**
 * Multiline still looks like it always
 * leaves the first line blank.
 */
```